recipes/dn.rb

```diff
@@ -89,6 +89,14 @@
   end
 end
 
+# include docker recipe if only on multi node,
+# otherwise we include hops::docker recipe in docker_registry which runs before hops::dn so we skip here
+unless exists_local("hops", "docker_registry")
```

Comment:
hops::dn recipe runs after kagent::install we including docker recipe here fixes registering when on data nodes. On single node, docker_registry runs before hops::dn so added check to avoid it running again